packages/now-build-utils/src/should-serve.ts

```diff
@@ -0,0 +1,26 @@
+import { parse } from 'path';
+import { ShouldServeOptions } from './types';
+
+export default function shouldServe({
+  entrypoint,
+  files,
+  requestPath
+}: ShouldServeOptions): boolean {
+  requestPath = requestPath.replace(/\/$/, ''); // sanitize trailing '/'
+  entrypoint = entrypoint.replace(/\\/, '/'); // windows compatibility
+
+  if (entrypoint === requestPath && hasProp(files, entrypoint)) {
+    return true;
+  }
+
+  const { dir, name } = parse(entrypoint);
+  if (name === 'index' && dir === requestPath && hasProp(files, entrypoint)) {
+    return true;
+  }
+
+  return false;
+}
+
+function hasProp (obj: Map<string, any>, key: string): boolean {
+  return Object.hasOwnProperty.call(obj, key)
```

Comment:
```suggestion
  return obj.has(key);
```

If this is truly a map, you can use has method. But then you don't really need `hasProp` helper at all 🤷‍♂️